src/App/App.js

```diff
@@ -1,66 +1,32 @@
 import React from 'react'
-import { HashRouter as Router, Route, Switch, Redirect } from 'react-router-dom'
-import { CssVariables } from '@dhis2/ui'
 
 import { DataStoreProvider } from '@dhis2/app-service-datastore'
 
-import {
-    Dashboard,
-    DataExportForExcel,
-    OutliersAndMissingDataAnalysis,
-    ConsistencyAnalysis,
-    AnnualReport,
-    Administration,
-} from '../sections'
+import { AppContextProvider } from './AppContext'
+import AppRouter from './AppRouter'
+import { ADMIN_AUTHORITIES } from '../config'
+import useAppDataQuery from '../hooks/useAppDataQuery'
 
-import { PATHS } from '../config'
-import SideNav from '../components/SideNav.js'
-import NoMatch from '../components/NoMatch.js'
-import classes from './App.module.css'
+const App = () => {
+    //TODO: take care of error
+    const { loading, data } = useAppDataQuery(ADMIN_AUTHORITIES)
 
-const App = () => (
-    <DataStoreProvider namespace="dataQualityTool">
-        <Router>
-            <CssVariables spacers colors />
-            <div className={classes.container}>
-                <aside className={classes.sidebar}>
-                    <SideNav />
-                </aside>
-                <main className={classes.content}>
-                    <Switch>
-                        <Route path={PATHS.DASHBOARD} component={Dashboard} />
-                        <Route
-                            exact
-                            path={PATHS.OUTLIERS_AND_MISSING_DATA_ANALYSIS}
-                            component={OutliersAndMissingDataAnalysis}
-                        />
-                        <Route
-                            exact
-                            path={PATHS.CONSISTENCY_ANALYSIS}
-                            component={ConsistencyAnalysis}
-                        />
-                        <Route
-                            exact
-                            path={PATHS.ANNUAL_REPORT}
-                            component={AnnualReport}
-                        />
-                        <Route
-                            exact
-                            path={PATHS.ADMINISTRATION}
-                            component={Administration}
-                        />
-                        <Route
-                            exact
-                            path={PATHS.DATA_EXPORT_FOR_EXCEL}
-                            component={DataExportForExcel}
-                        />
-                        <Redirect exact from="/" to={PATHS.DASHBOARD} />
-                        <Route component={NoMatch} />
-                    </Switch>
-                </main>
-            </div>
-        </Router>
-    </DataStoreProvider>
-)
+    if (loading) {
+        //TODO: add loading component when we implemented the deferred loading spinner with 700ms delay in @dhis2/ui
+        return null
+    }
+
+    const currentUserIsAdmin = data.userAuthorities.some(auth =>
+        ADMIN_AUTHORITIES.includes(auth)
+    )
```

Comment:
Probably still use conditional chaining here (`data?.userAuthorities`), as I think data should also keep an `undefined` state. See next comment.

Reply:
added a check for error instead before this line, assuming data is what we expect when we get here.